File: src/main/java/com/travel/marketplace/modules/review/service/ReviewService.java
package com.travel.marketplace.modules.review.service;

import com.travel.marketplace.modules.review.dto.ReviewCreateRequest;
import com.travel.marketplace.modules.review.dto.ReviewResponse;
import com.travel.marketplace.modules.review.dto.ReviewSummaryResponse;
import com.travel.marketplace.modules.review.dto.ReviewUpdateRequest;
import com.travel.marketplace.modules.review.enums.ReviewStatus;
import com.travel.marketplace.modules.review.enums.TripType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ReviewService {
    Page<ReviewResponse> getListingReviews(Long listingId, Integer rating, TripType tripType, Pageable pageable);
    ReviewSummaryResponse getListingReviewSummary(Long listingId);
    ReviewResponse createReview(Long listingId, Long userId, ReviewCreateRequest request);
    ReviewResponse updateReview(Long reviewId, Long userId, ReviewUpdateRequest request);
    void deleteReview(Long reviewId, Long userId);
    Page<ReviewResponse> getReviewsForAdmin(ReviewStatus status, Long listingId, Pageable pageable);
    ReviewResponse hideReview(Long reviewId);
    ReviewResponse publishReview(Long reviewId);
}

File: src/services/reviewService.ts
import api from './api';
import { ApiResponse, PageResponse } from '@/types';
import {
  Review,
  ReviewCreateRequest,
  ReviewSearchParams,
  ReviewSummary,
  ReviewUpdateRequest,
} from '@/types/review';

export const reviewService = {
  getListingReviews: async (listingId: number, params?: ReviewSearchParams) => {
    const response = await api.get<ApiResponse<PageResponse<Review>>>(`/listings/${listingId}/reviews`, { params });
    return response.data;
  },

  getListingReviewSummary: async (listingId: number) => {
    const response = await api.get<ApiResponse<ReviewSummary>>(`/listings/${listingId}/reviews/summary`);
    return response.data;
  },

  createReview: async (listingId: number, data: ReviewCreateRequest) => {
    const response = await api.post<ApiResponse<Review>>(`/listings/${listingId}/reviews`, data);
    return response.data;
  },

  updateReview: async (reviewId: number, data: ReviewUpdateRequest) => {
    const response = await api.put<ApiResponse<Review>>(`/reviews/${reviewId}`, data);
    return response.data;
  },

  deleteReview: async (reviewId: number) => {
    const response = await api.delete<ApiResponse<void>>(`/reviews/${reviewId}`);
    return response.data;
  },
};
